src/main/resources/native/android/android_project/app/src/main/java/com/gluonhq/helloandroid/NativeWebView.java

```diff
@@ -0,0 +1,252 @@
+                    }
+
+                    @Override
+                    public void onReceivedHttpError(WebView view, WebResourceRequest request, WebResourceResponse errorResponse) {
+                        Log.v(TAG, "LOAD onReceivedHttpError: request: " + request.getMethod());
+                        Log.v(TAG, "LOAD onReceivedHttpError: errorResponse: " + errorResponse.getReasonPhrase());
+                        nativeFailedURL(request.getUrl().toString());
+                    }
+
+                    @Override
+                    public void onReceivedSslError(WebView view, SslErrorHandler handler, SslError error) {
+                        Log.v(TAG, "onReceivedSslError, skipping error: " + error);
+                        handler.proceed();
+                    }
+                });
+
+                WebSettings webSettings = NativeWebView.this.webView.getSettings();
+                // TODO, pass from Java
+                webSettings.setJavaScriptEnabled(true);
+                webSettings.setJavaScriptCanOpenWindowsAutomatically(true);
+                webSettings.setDomStorageEnabled(true);
+                webSettings.setUseWideViewPort(true);
+                webSettings.setLoadWithOverviewMode(true);
+
+                Log.v(TAG, "NATIVEWEBVIEW wv = "+NativeWebView.this.webView);
+                Log.v(TAG, "finally, NATIVEWEBVIEW wv = "+NativeWebView.this.webView);
+            }
+        });
+        reLayout();
+        Log.v(TAG, "NATIVEWEBVIEW constructor returns: "+this);
+    }
+
+    public void loadUrl(final String url) {
+        Log.v(TAG, "in dalvik, loadUrl called wwith url = "+url+" and webView = "+this.webView);
+        instance.runOnUiThread(new Runnable () {
+            public void run() {
+                NativeWebView.this.webView.loadUrl(url);
+            }
+        });
+    }
+
+    public String executeScript(final String script) {
+        Log.v(TAG, "in dalvik, loadUrl called with script  and webView = "+this.webView);
+        final CountDownLatch latch = new CountDownLatch(1);
+        scriptResult = null;
+        Runnable action = new Runnable() {
+            @Override
+            public void run() {
+                NativeWebView.this.webView.evaluateJavascript(script, new ValueCallback<String>() {
+                    @Override
+                    public void onReceiveValue(String s) {
+                        Log.v(TAG, "in dalvik, script result: " + s);
+                        scriptResult = s;
+                        latch.countDown();
+                    }
+                });
+            }
+        };
+        webView.post(action);
+        try {
+            latch.await();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        Log.v(TAG, "in dalvik, loadUrl script result = " + scriptResult);
+        return scriptResult;
+    }
+
+    private void setVisible(boolean visible) {
```

Comment:
Codacy found an issue: [Avoid unused private methods such as 'setVisible(boolean)'.](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=6369267)

Reply:
Codacy is dropping the ball here, as it is used (via JNI)